storage: covert SizeSpec to protobuf

andrewbaptist opened the pull request:
This commit moves SizeSpec to the storagepb directory and converts it toa protobuf. Additionally it pulls shared storage into the global StorageConfig file.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140071)
<!-- Reviewable:end -->

Commit:
server: split size_spec off from store_spec

This commit splits these types into separate files which helps with code
cleanup and re-org.

Epic: none

Release note: None

Commit:
storage: covert SizeSpec to protobuf

This commit moves SizeSpec to the storagepb directory and converts it to
a protobuf. Additionally it pulls shared storage into the global
StorageConfig file.

Epic: none

Release note: None

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/140071#-:-OHx8rd8DIZQvdm8fBio:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andrewbaptist, @jbowens, and @sumeerbhola)

___
*[`pkg/storage/storagepb/storage.proto` line 87 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/140071#-OHx7s5N0ns_Khdp9gk2:-OHx7s5N0ns_Khdp9gk3:b9j08y9) ([raw file](https://github.com/cockroachdb/cockroach/blob/75548fc028df9d845254db11ecfdf16fb5925ea9/pkg/storage/storagepb/storage.proto#L87)):*
> ```proto
> message SizeSpec {
>   option (gogoproto.goproto_stringer) = false;
>   // capacity is how much space on the file system to use. if 0 then use the
> ```

[nit] Capacity, Percent


<!-- Sent from Reviewable.io -->

andrewbaptist commented:
TFTR!

bors r=RaduBerinde
